scripts/lib/CIME/namelist.py

```diff
@@ -112,7 +112,8 @@
 
 # Fortran syntax regular expressions.
 # Variable names.
-FORTRAN_NAME_REGEX = re.compile(r"^[a-z][a-z0-9_]{0,62}$", re.IGNORECASE)
+FORTRAN_NAME_REGEX = re.compile(r"(^[a-z][a-z0-9_]{0,62})(\(([+-]?\d*:?){3}\))?$", re.IGNORECASE)
```

Comment:
This regex has issues. For instance, it happily accepts:
`foo(-:2:1:)`
which has two issues. One is the third colon and the other is the allowance of a sign with no digits (which makes Intel choke but surprisingly, nag doesn't mind).
I have a regex that works (at least as far as I can tell):
`re.compile(r"(^[a-z][a-z0-9_]{0,62})((\(([+-]?\d+)\))|(\(([+-]?\d+)?:([+-]?\d+)?:?([+-]?\d+)?\)))?$", re.IGNORECASE)`
It seems a bit  clumsy but has the advantage that you can recover each item in a subsection (group(6-8)).